build.gradle

```diff
@@ -254,18 +256,18 @@ configure(javaProjects) {
     }
 
     // Enable JaCoCo test coverage when '-Pcoverage' option is specified.
-    def jacocoEnabled = project.hasProperty('coverage')
-
-    test {
+    tasks.withType(Test) {
         jacoco {
-            enabled = jacocoEnabled
+            enabled = project.ext.jacocoEnabled
             append = false
         }
     }
 
     jacocoTestReport {
         reports {
-            xml.enabled jacocoEnabled
+            csv.enabled false
+            xml.enabled true
+            html.enabled true
```

Comment:
I realized I sometimes want to generate test coverage report from the previous run. There was really no need to disable it when `-Pcoverage` is not specified.